### PR TITLE
Hide robin hood

### LINKED ISSUE #2445: Valhalla "leaks" dependence on robin-hood?
AFAICT, the only place RHH is used "publicly" is in [`thor/cost_matrix.h`](https://github.com/valhalla/valhalla/blob/0f160fdfe8c16f7ca5508445bf09fdef8ecaa073/valhalla/thor/costmatrix.h#L161)... and it's not a public member.

This seems to be unnecessarily leaking an implementation detail and making it unnecessarily impossible for external consumers to use Valhalla without also dragging in a dependency on RHH.

Would it be feasible to "hide" this dependency in some manner, possibly by using [PIMPL](https://en.wikipedia.org/wiki/Opaque_pointer) on this class?

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Modify `thor::CostMatrix` to make the target map an opaque type. Make `idtable.h` explicitly a private header (by moving it to `src/`) and remove its inclusion from `pbfgraphparser.cc`, which was not actually using it.

This removes all instances of a public (or "public", in the case of `idtable.h`) header including `robin_hood.h`, such that our use thereof is totally hidden to consumers rather than leaked into the public interface. In particular, this means that consumers no longer need to be able to include the RHH headers. (Which is very fortunate, as we weren't installing them!)

Fixes #2445.

### CODE REVIEW

File: src/thor/costmatrix.cc [hunk 1]
```diff
@@ -6,6 +6,8 @@
 #include "thor/costmatrix.h"
 #include "worker.h"
 
+#include <robin_hood.h>
```
Comment (mandeepsandhu) [inline src/thor/costmatrix.cc:9]: I dont think this is a system installed header, in which case its canonical to use `#include "robin_hood.h"` (instead of the angled braces).

File: src/mjolnir/pbfadminparser.cc [hunk 2]
```diff
@@ -2,8 +2,8 @@
 #include "mjolnir/pbfadminparser.h"
 #include "admin_lua_proc.h"
 #include "baldr/tilehierarchy.h"
+#include "idtable.h"
```
Comment (kevinkreiser) [inline src/mjolnir/pbfadminparser.cc:5]: This change isn't needed can you please revert

File: src/thor/costmatrix.cc (hunk 1 above)
Comment (kevinkreiser) [reply src/thor/costmatrix.cc:9]: We use cmake to import thir party header directories essentially with the `-I` flag. The angle brackets are correct because of this

File: test/idtable.cc [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#include "mjolnir/idtable.h"
+#include "../src/mjolnir/idtable.h"
```
Comment (kevinkreiser) [inline test/idtable.cc:2]: You can also revert the changes in this file

**Comment by kevinkreiser** [comment]
Other than the unneeded changes this looks good

File: src/mjolnir/pbfadminparser.cc (hunk 2 above)
Comment (mwoehlke-kitware) [reply src/mjolnir/pbfadminparser.cc:5]: Huh?

```
valhalla/src/mjolnir/pbfadminparser.cc: At global scope:
valhalla/src/mjolnir/pbfadminparser.cc:134:3: error: ‘UnorderedIdTable’ does not name a type
  134 |   UnorderedIdTable shape_, members_;
      |   ^~~~~~~~~~~~~~~~
```

That include is needed. If you meant where it appears, that is required by `clang-format`.

File: test/idtable.cc (hunk 3 above)
Comment (mwoehlke-kitware) [reply test/idtable.cc:2]: No, I can't; if I revert it, the header can't be found, and CI fails.

**Comment by mwoehlke-kitware** [comment]
> Other than the unneeded changes [...]

I don't understand what "unneeded changes" you mean. Both changes on which you commented were not present in earlier versions of this PR, which failed CI.

File: test/idtable.cc (hunk 3 above)
Comment (mwoehlke-kitware) [reply test/idtable.cc:2]: Specifically:
```
valhalla/test/idtable.cc:1:10: fatal error: mjolnir/idtable.h: No such file or directory
    1 | #include "mjolnir/idtable.h"
```

File: src/mjolnir/pbfadminparser.cc (hunk 2 above)
Comment (mwoehlke-kitware) [reply src/mjolnir/pbfadminparser.cc:5]: If you meant removing the `mjolnir/`, that is consistent with how other "private" headers (and there are several) are included.

Comment (kevinkreiser) [reply src/mjolnir/pbfadminparser.cc:5]: i completely missed this part of the diff:

`valhalla/mjolnir/idtable.h → src/mjolnir/idtable.h`

File: test/idtable.cc (hunk 3 above)
Comment (kevinkreiser) [reply test/idtable.cc:2]: same as above, i didnt see you actually moved the header to the source directory. though as i mentioned before i wonder why it even makes sense to have this class as a separate header at this point

**Comment by mwoehlke-kitware** [comment]
> sorry i missed the single line part of the diff where it showed you moved the idtable header to the source directory tree

Hehe... yeah, that would do it :smile:. No worries, I can totally understand that happening.

File: test/idtable.cc (hunk 3 above)
Comment (mwoehlke-kitware) [reply test/idtable.cc:2]: > I wonder why it even makes sense to have this class as a separate header at this point

The short answer is that I'm not sure how the unit test is supposed to work otherwise. If you have suggestions...?
